src/vsphere_cpi/spec/integration/vm_host_rule_spec.rb

```diff
@@ -160,32 +168,52 @@
               'clusters' => [
                 {
                   @cluster_name => {
-                    'host_group' =>  @second_host_group,
+                    'host_group' =>  {'name' => @second_host_group, 'drs_rule' => drs_rule}
                   },
                 },
                 {
                   @cluster_name => {
-                    'host_group' =>  @first_host_group,
+                    'host_group' =>  {'name' => @first_host_group, 'drs_rule' => drs_rule}
                   }
                 },
               ]
             }]
           }
         end
-        it 'creates VM and attach vm group host affinity rule' do
-          simple_vm_lifecycle(cpi, '', vm_type, get_network_spec) do |vm_id|
-            vm = cpi.vm_provider.find(vm_id)
-            expect(vm.cluster).to eq(@cluster_name)
-            # We expect VM to be created on first host group
-            # because first host group has two hosts and more memory.
-            expect(hosts_in_first_host_group).to include(vm.mob.runtime.host.name)
-            expect(get_count_vm_host_affinity_rules(cluster_mob)).to eq(1)
-            expect(find_vm_group_mob(@first_host_group).vm).to include(vm.mob)
-            vm_host_rule = get_vm_host_affinity_rule(cluster_mob)
-            expect(vm_host_rule.vm_group_name).to eq(first_host_vm_group_name)
-            expect(vm_host_rule.affine_host_group_name).to eq(@first_host_group)
+        context 'when the drs rule type is must' do
+          it 'creates VM and attach vm group host affinity rule' do
+            simple_vm_lifecycle(cpi, '', vm_type, get_network_spec) do |vm_id|
+              vm = cpi.vm_provider.find(vm_id)
+              expect(vm.cluster).to eq(@cluster_name)
+              # We expect VM to be created on first host group
+              # because first host group has two hosts and more memory.
+              expect(hosts_in_first_host_group).to include(vm.mob.runtime.host.name)
+              expect(get_count_vm_host_affinity_rules(cluster_mob)).to eq(1)
+              expect(find_vm_group_mob(@first_host_group).vm).to include(vm.mob)
+              vm_host_rule = get_vm_host_affinity_rule(cluster_mob, drs_rule)
+              expect(vm_host_rule.vm_group_name).to eq(first_host_vm_group_name)
+              expect(vm_host_rule.affine_host_group_name).to eq(@first_host_group)
+            end
+            expect(get_count_vm_host_affinity_rules(cluster_mob)).to eq(0)
+          end
+        end
+        context 'when the drs rule type is should' do
+          let(:drs_rule) { 'SHOULD' }
+          it 'creates VM and attach vm group host affinity rule' do
```

Comment:
This may benefit from RSpec's shared examples as I can't tell how it differs (other than the `'SHOULD'`) from the above `it` example block:

https://relishapp.com/rspec/rspec-core/docs/example-groups/shared-examples